### PR TITLE
Us s4 article creation

### PR DESCRIPTION
**Pull request by alexboutemy**

### CODE REVIEW

**Review by pwah59 — DISMISSED** [review]

**Review by pwah59 — APPROVED** [review]

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/pages/creation-article/creation-article.component.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  templateUrl: './creation-article.component.html',
+  styleUrls: ['./creation-article.component.scss']
+})
+export class CreationArticleComponent implements OnInit {
+  
+  article!:Article;
+
+  user = this.userService.getUser() as User;
+
+  categories!:any;
```
[the remark this answers is not in the document]
Comment (alexboutemy) [reply src/app/pages/creation-article/creation-article.component.ts:20]: changed to Category[]

File: src/app/pages/creation-article/creation-article.component.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      .replace(/[\s_-]+/g, '-')
+      .replace(/^-+|-+$/g, '');
+    }
+
+  duplicateContent(event: any) {
+    const iconElement = event.currentTarget as HTMLElement;
+    const contentDiv = iconElement.closest('.content') as HTMLElement;
+    const clonedContent = contentDiv.cloneNode(true) as HTMLElement;
+
+    // Réinitialiser les champs de texte dans la copie dupliquée
```
[the remark this answers is not in the document]
Comment (alexboutemy) [reply src/app/pages/creation-article/creation-article.component.ts]: done

File: src/app/pages/creation-article/creation-article.component.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    clonedContent.querySelectorAll('input[type="text"]').forEach((input: any) => {
+        input.value = '';
+    });
+
+    clonedContent.querySelectorAll('textarea').forEach((input: any) => {
+      input.value = '';
+  });
+    // Supprimer l'écouteur de clic de l'icône "plus" cloné pour éviter les duplications répétées
```
[the remark this answers is not in the document]
Comment (alexboutemy) [reply src/app/pages/creation-article/creation-article.component.ts]: done

File: src/app/pages/creation-article/creation-article.component.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    const clonedIcon = clonedContent.querySelector('.fa-plus');
+    clonedIcon?.removeEventListener('click', this.duplicateContent);
+
+    // Ajouter l'écouteur de clic à l'icône "plus" du nouveau contenu cloné
```
[the remark this answers is not in the document]
Comment (alexboutemy) [reply src/app/pages/creation-article/creation-article.component.ts]: done

File: src/app/pages/creation-article/creation-article.component.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    clonedIcon?.addEventListener('click', (event) => this.duplicateContent(event));
+
+    // Ajouter le contenu dupliqué après la balise d'origine
```
[the remark this answers is not in the document]
Comment (alexboutemy) [reply src/app/pages/creation-article/creation-article.component.ts]: done

**Review by pwah59 — APPROVED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**